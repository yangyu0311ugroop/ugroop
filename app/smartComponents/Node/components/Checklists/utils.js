import { CLOSED, COMPLETED, OPEN, OUTSTANDING } from 'appConstants';
import { CONSOLE_HELPERS, MISSING_REQUIRED_PROPS } from 'utils/helpers/console';
import { LOGIC_HELPERS } from 'utils/helpers/logic';
import { CHECKITEM, CHECKLIST } from 'utils/modelConstants';

const toggleStatus = (params, props) => {
  const { customData, type, status, ...node } = params;
  const { me } = props;

  let nextStatus;
  let newCustomData;

  switch (type) {
    case CHECKITEM:
      nextStatus = LOGIC_HELPERS.ifElse(
        status === COMPLETED,
        OUTSTANDING,
        COMPLETED,
      );
      newCustomData = LOGIC_HELPERS.ifElse(
        nextStatus === COMPLETED,
        {
          completedAt: new Date().toISOString(),
          completedBy: me,
        },
        {
          completedAt: null,
          completedBy: null,
        },
      );
      break;
    case CHECKLIST:
      nextStatus = LOGIC_HELPERS.ifElse(status === CLOSED, OPEN, CLOSED);
      newCustomData = LOGIC_HELPERS.ifElse(
        nextStatus === CLOSED,
        {
          completedAt: new Date().toISOString(),
          completedBy: me,
        },
        {
          completedAt: null,
          completedBy: null,
        },
      );
      break;
    default:
      CONSOLE_HELPERS.error(
        MISSING_REQUIRED_PROPS,
        'CHECKLIST_HELPERS.toggleStatus: type',
      );
      return null;
  }

  return {
    ...node,
    customData: { ...customData, ...newCustomData },
    type,
    status: nextStatus,
  };
};

export const CHECKLIST_HELPERS = {
  toggleStatus,
};
