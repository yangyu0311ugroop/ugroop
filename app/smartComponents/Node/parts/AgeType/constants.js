export const AGE_TYPE_OPTIONS = {
  Adult: 'Adult',
  Child: 'Child',
};
