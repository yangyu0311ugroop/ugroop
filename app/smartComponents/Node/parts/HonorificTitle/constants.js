export const TITLE_OPTIONS = {
  Mr: 'Mr',
  Mrs: 'Mrs',
  Miss: 'Miss',
  Ms: 'Ms',
  Sir: 'Sir',
  Dr: 'Dr',
  Others: 'Other',
};
