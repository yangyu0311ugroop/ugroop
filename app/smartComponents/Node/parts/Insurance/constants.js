export const INSURANCE_MODES = {
  pending: 'pending',
  confirmed: 'confirmed',
  other: 'other',
};
