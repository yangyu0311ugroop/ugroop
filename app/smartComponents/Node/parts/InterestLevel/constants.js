export const INTEREST_LEVELS = {
  interested: 'interested',
  reserve: 'reserve',
  ready: 'ready',
};
