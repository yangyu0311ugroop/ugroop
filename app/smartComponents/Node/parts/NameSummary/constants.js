export const OVERVIEW_FORMATS = ['bold', 'italic', 'color'];
