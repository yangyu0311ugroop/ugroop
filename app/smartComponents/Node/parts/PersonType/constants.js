export const PERSON_TYPES = {
  student: 'student',
  parent: 'parent',
  teacher: 'teacher',
  leader: 'leader',
  none: '',
};

export const EDUCATIONAL_PERSON_TYPES = [
  'student',
  'parent',
  'teacher',
  'leader',
];
