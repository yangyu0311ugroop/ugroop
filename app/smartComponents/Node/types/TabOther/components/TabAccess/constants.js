export const PUBLIC = 'public';
export const ORGANISERS = 'organisers';
export const ONLY_ME = 'onlyMe';
export const SHOW = 'show';
export const HIDE = 'hide';

export const ACCESS_LEVEL = {
  [PUBLIC]: {
    icon: 'lnr-earth',
    title: 'Everyone',
    longTitle: 'Shared with Everyone',
    description: 'Everyone in this tour',
  },
  [ORGANISERS]: {
    icon: 'lnr-users',
    title: 'Organisers',
    longTitle: 'Shared with Organisers',
    description: 'Tour owner and organisers',
  },
  [ONLY_ME]: {
    icon: 'lnr-lock',
    title: 'Private',
    longTitle: 'Private',
    description: 'Only me',
  },
};
export const PRINT_MODE = {
  [SHOW]: {
    icon: 'lnr-printer',
    title: 'Show',
    longTitle: 'Will show on print view',
    description: 'Will show on print view',
  },
  [HIDE]: {
    icon: 'lnr-layers-crossed',
    title: 'Hide',
    longTitle: 'Hidden on print view',
    description: 'Hidden on print view',
  },
};
export const ACCESS_VARIANTS = {
  sharedWith: 'sharedWith',
  printMode: 'printMode',
};

ACCESS_LEVEL.null = ACCESS_LEVEL[PUBLIC]; // default
ACCESS_LEVEL[''] = ACCESS_LEVEL[PUBLIC]; // default
PRINT_MODE.null = PRINT_MODE[SHOW]; // default
PRINT_MODE[''] = PRINT_MODE[SHOW]; // default
