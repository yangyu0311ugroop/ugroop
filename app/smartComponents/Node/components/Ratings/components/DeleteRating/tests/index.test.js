import { NODE_API, REMOVE_NODE_AND_LINKS } from 'apis/constants';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import { RATINGS } from 'utils/modelConstants';
import { DeleteRating } from '../index';

describe('<DeleteRating />', () => {
  let rendered;
  let instance;

  const resaga = {
    setValue: jest.fn(),
    dispatchTo: jest.fn(),
  };

  const props = {
    classes: {},
    resaga,
  };

  beforeEach(() => {
    rendered = shallow(<DeleteRating {...props} />);
    instance = rendered.instance();
  });

  it('should exists', () => {
    expect(DeleteRating).toBeDefined();
  });

  it('should render without exploding', () => {
    expect(rendered.length).toBe(1);
  });

  describe('handleRemoveSuccess', () => {
    it('should call onClose', () => {
      const onClose = jest.fn();
      rendered.setProps({ onClose });

      instance.handleRemoveSuccess();

      expect(onClose).toBeCalled();
    });
  });

  describe('handleClick', () => {
    it('should call dispatch to', () => {
      rendered.setProps({
        id: 1,
        participantId: 2,
        parentNodeId: 3,
      });
      instance.handleRemoveSuccess = 'handleRemoveSuccess';
      instance.handleClick();

      expect(resaga.dispatchTo).toBeCalledWith(
        NODE_API,
        REMOVE_NODE_AND_LINKS,
        {
          payload: {
            id: 1,
            fk: 2,
            parentNodeId: 3,
            childKey: RATINGS,
            nextNodeChildKey: RATINGS,
          },
          onSuccess: 'handleRemoveSuccess',
        },
      );
    });
  });

  describe('render()', () => {
    it('should render correctly', () => {
      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });
});
