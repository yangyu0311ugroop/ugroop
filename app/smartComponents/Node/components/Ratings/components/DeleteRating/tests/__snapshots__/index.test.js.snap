// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteRating /> render() should render correctly 1`] = `
<div>
  <WithStyles(Button)
    color="alert"
    dense={true}
    loading={false}
    onClick={[Function]}
  >
    <FormattedMessage
      defaultMessage="Delete"
      id="app/smartComponents/Node/components/Ratings/components/DeleteRating/button"
    />
  </WithStyles(Button)>
</div>
`;
